From 3b8ee49557017fb72717d2c2764f8f699834639e Mon Sep 17 00:00:00 2001
From: Biju Chakkachattil <bijux.chakkachattil@intel.com>
Date: Mon, 11 May 2015 10:52:39 +0530
Subject: [PATCH] [WA]EM: Disable OTG in Battery Saver Mode

The cable state is notified through extcon USB
notification which is used to turn on/off vbus.

This is a WA patch, since GPIO changes for driving
vbus in host mode are not yet supported in BIOS.

Change-Id: Ie50ef6ffa8219a11ed994761f687c97aadd6ca1f
Tracked-On: https://jira01.devtools.intel.com/browse/AKP-1596
Signed-off-by: Biju Chakkachattil <bijux.chakkachattil@intel.com>
---
 drivers/power/intel_pmic_ccsm.c | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/drivers/power/intel_pmic_ccsm.c b/drivers/power/intel_pmic_ccsm.c
index a432489fa1a0..511292c18c03 100644
--- a/drivers/power/intel_pmic_ccsm.c
+++ b/drivers/power/intel_pmic_ccsm.c
@@ -1583,9 +1583,12 @@ static int vbus_set_cur_state(struct thermal_cooling_device *tcd,
 	/**
 	 * notify directly only when the ID_GND and want to change the state
 	 * from previous state (vbus enable/disable).
+	 * Otherwise, check cable_state to determine OTG connect/disconnect
+	 * status based on USB notification and enable/disable vbus.
 	 */
 	mutex_lock(&pmic_lock);
-	if ((pmic_get_usbid() == RID_GND) && (chc.vbus_state != new_state)) {
+	if (((pmic_get_usbid() == RID_GND) || chc.cable_state) &&
+		(chc.vbus_state != new_state)) {
 		if (!new_state) {
 			if (chc.otg->set_vbus)
 				chc.otg->set_vbus(chc.otg, true);
-- 
1.9.1

